torch/utils/data/dataloader.py

```diff
@@ -229,7 +229,8 @@ def __init__(self, dataset: Dataset[T_co], batch_size: Optional[int] = 1,
             # this, and support custom samplers that specify the assignments to
             # specific workers.
             if isinstance(dataset, IterDataPipe):
-                dataset = torch.utils.data.graph_settings.apply_shuffle_settings(dataset, shuffle=shuffle)
+                if shuffle is not None:
+                    dataset = torch.utils.data.graph_settings.apply_shuffle_settings(dataset, shuffle=shuffle)
             elif shuffle is not False:
                 raise ValueError(
                     "DataLoader with IterableDataset: expected unspecified "
```

Comment:
Could we check `elif shuffle is not False or shuffle is not None:` here to prevent BC breaking.

Reply:
I thought you wanted 100% BC. With your suggestion passing `shuffle=None` will result in no error, but it currently does.

Reply:
I think my proposed way is BC because we change the default value of `shuffle`. For users who use `IterableDataset` with default `shuffle` before, there won't be any Error. But, it currently does raise Error.

Reply:
Take the following example:

```py
from torch.utils.data import DataLoader, IterableDataset


class Dataset(IterableDataset):
    def __iter__(self):
        return
        yield
```

- on `master`:

    ```py
    DataLoader(Dataset(), shuffle=False)  # passes
    DataLoader(Dataset(), shuffle=None)  # fails
    ```

- this PR

    ```py
    DataLoader(Dataset(), shuffle=False)  # passes
    DataLoader(Dataset(), shuffle=None)  # fails
    ```

- this PR with your suggestion

    ```py
    DataLoader(Dataset(), shuffle=False)  # fails
    DataLoader(Dataset(), shuffle=None)  # fails
    ```

Reply:
I mean:
```py
DataLoader(Dataset())  # passes on master
DataLoader(Dataset())  # fails on this PR
```

Reply:
That being said, there are failing tests related to this, that I need to take care of.